### ISSUE TITLE
Unsure how to use magic commands to display the output of a code block

### ISSUE THREAD

**Issue by ngirard** [twibiral/obsidian-execute-code#232]
Heya,

In a similar vein to #210, I have no idea after grasping the README how a png or HTML output produced by a code block can be displayed using the `@show(ImagePath)` and `@html(HtmlSource)` magic commands, if ever possible.

I'd suggest considering allowing a postfix notation for these "magic commands", which would allow them to be composable.
Postfix notation is a frequent choice in this context — out of my mind, I'm thinking of Mathematica or the display operators provided by the various IPython/Jupyter kernels (see for instance `displayHTML` for the [Bash kernel](https://github.com/takluyver/bash_kernel)).

Cheers, and thank you for involving into this nice project !

**Comment by twibiral** [comment]
Hi @ngirard, you don't need `show` for outputting html. You can just put arbitrary html code in the `html` command like this:

```python
print("Hello World") # arbitrary Python code

html("<div align='center'>Hello World</div>")
```

Does this resolve your issue?

**Comment by twibiral** [comment]
Hi @ngirard, you don't need `show` for outputting html. You can just put arbitrary html code in the `html` command like this:

```python
print("Hello World") # arbitrary Python code

html("<div align='center'>Hello World</div>")
```

Does this resolve your issue?

**Comment by ngirard** [comment]
Not quite.
Think of a block of code that produces HTML, say
```py
my_python_code_that_produces_html(with, args)
```
How to display that output using `@html` ?

Please follow my link above to get a grasp for what I mean.

**Comment by twibiral** [comment]
You only linked a whole Repo, I don't know what to look for in there...

If your function returns a string of html, than you can simply write:

```python
@html(yourFunc(a, b, c))
```

**Comment by ngirard** [comment]
> You only linked a whole Repo, I don't know what to look for in there...

I specifically mentioned postfix notation and `displayHTML` as an example.

![image](https://user-images.githubusercontent.com/98098/221184406-5d866de0-d61c-476f-b25f-37ff67f77b6b.png)

**Comment by twibiral** [comment]
Then the answer is that we will not introduce bash syntax into python. Maybe we add it for shell commands.

The example doesn't do anything, that can't be done with the existing syntax...

**Comment by ngirard** [comment]
> The example doesn't do anything, that can't be done with the existing syntax...

I'd beg to differ. It can be argued than Mathematica's success as a language comes from its support for both postfix and infix notation. Composability matters...

Thanks for your feedback and consideration !

**Comment by twibiral** [comment]
Yes, but our goal wasn't to create a new programming language but to enable the use of existing languages in Obsidian. You can use mathematica with the plugin by the way
